lib/matplotlib/axes/_base.py

```diff
@@ -417,16 +458,16 @@ def _plot_args(self, tup, kwargs, return_kwargs=False):
             raise ValueError("x, y, and format string must not be None")
```

Comment:
This would never have checked whether `fmt` is None, because of the first if, is it not?

Reply:
For `(x, y, None)`, you ended up with the "third arg must be a format string" error. But for `(y, None)` this check triggers.

Reply:
Ah yes, because the length checks are different.